Add support for Vulkan on Wayland

myfreeweb opened the pull request:
Works on `QT_QPA_PLATFORM=wayland-egl`:

![rpcs3-wayland-vulkan](https://user-images.githubusercontent.com/208340/34389127-f4ce85f8-eb48-11e7-9487-e8d8ebf5267c.png)

Megamouse commented:
please try to adjust to our coding style as described in here: https://github.com/RPCS3/rpcs3/wiki/Coding-Style
only then we can properly review this contribution.

There are a lot of missing or wrong indentations like with the namespaces and you often have opening braces in the same line instead of a new one.

kd-11 commented on rpcs3/rpcs3qt/gs_frame.cpp:
```diff
@@ -145,6 +150,17 @@ void gs_frame::show()
 
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
+std::pair<wl_display*, wl_surface*> gs_frame::wayland_handle() const
+{
```
Move this into gs_frame::handle(). The checks in VKGSRender to check if wayland handle is valid can also be done there.

kd-11 commented on rpcs3/Emu/RSX/VK/VKHelpers.h:
```diff
@@ -1309,16 +1317,36 @@ namespace vk
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
+			, std::pair<wl_display*, wl_surface*>
+#endif
+		> VkUnixContext;
```
This template can be moved to gsframe header instead as it is not restricted to use with vulkan only and holds general unix window and display properties. Name can be changed to reflect that as well, something like unix_display_context or something along those lines

myfreeweb commented:
Moved the stuff to gsframe header. New version: `handle()` returns `display_handle_t` which an alias for `HWND` on Windows, and this `variant` on Unix.

also moved some identical code outside of platform ifdefs in VKGSRender.cpp

kd-11 commented on rpcs3/rpcs3qt/gs_frame.cpp:
```diff
@@ -8,6 +8,13 @@
 #include <QTimer>
 #include <QThread>
 
+#ifdef VK_USE_PLATFORM_WAYLAND_KHR
```
This whole block from line 11 to 16 should be in an ifndef _WIN32 block. That should fix appveyor CI and allow this set to be mergeable

JXPZ commented:
![esboco](https://user-images.githubusercontent.com/35011686/34728938-713b8c96-f542-11e7-936a-ed6a679908d2.png) Drive Geforce 390.65

JXPZ commented:
1050 TI OC EDITION 4GB GIGABYTE

Inviuz commented:
@JXPZ Completely unrelated. The issue that talks about problems with newest nvidia driver is already closed. Issue number #4036
